### PR TITLE
[Process] Implement IteratorAggregate to stream output

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | symfony/symfony-docs#6424 |

Sibling to #18386 for iterating of the output streams.

### CODE REVIEW

**Comment by romainneutron** [comment]
Process is getting some freshness :+1:

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by HeahDude** [comment]
The Process CHANGELOG needs that freshness :smiley:

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -499,6 +499,54 @@ public function getIncrementalOutput()
...
+     *
+     * @param bool $blocking    Whether to use a blocking read call.
+     * @param bool $clearOutput Whether to clear or keep output in memory.
+     *
+     * @throws LogicException in case the output has been disabled
+     * @throws LogicException In case the process is not started
+     *
+     * @return \Generator
+     */
+    public function getIterator($blocking = true, $clearOutput = true)
```
Comment (stof) [inline src/Symfony/Component/Process/Process.php]: This signature does not match the ``\IteratorAggregate``, which looks weird to me.

**Pull request merged**